homeassistant/components/media_player/cambridgeaudio.py

```diff
@@ -0,0 +1,382 @@
+        if self._audio_source == "media player":
+            self._smdevice.trnsprt_next()
+
+    def media_previous_track(self):
+        """Skip to the previous track, when in media player mode."""
+        if self._audio_source == "media player":
+            self._smdevice.trnsprt_prev()
+
+    def media_pause(self):
+        """Pause playing the current media."""
+        # there is no "pause" with internet radio streams
+        if self._audio_source == "media player":
+            self._smdevice.trnsprt_pause()
+            self._state = STATE_PAUSED
+        elif self._audio_source == "internet radio":
+            self.media_stop()
+            self._state = STATE_IDLE
+
+    def media_seek(self, position):
+        """Jump to the specified percent position within the current track."""
+        abs_pos = round(self._to_seconds(self._duration) * (position/100), 2)
+        abs_pos_h = int(abs_pos / 3600)
+        abs_pos_m = int((abs_pos % 3600) / 60)
+        abs_pos_s = int((abs_pos % 3600) % 60)
+        abs_pos = '{:01d}:{:02d}:{:02d}'.format(abs_pos_h,
+                                                abs_pos_m,
+                                                abs_pos_s)
+        self._smdevice.trnsprt_seek(abs_pos)
+
+    def media_stop(self):
+        """Stop playing the current media."""
+        self._smdevice.trnsprt_stop()
+
+    def media_play(self):
+        """Start playing the current media."""
+        self._smdevice.trnsprt_play()
+        self._state = STATE_PLAYING
+
+    def mute_volume(self, mute):
+        """Mute the volume. This only works in pre-amp mode."""
+        self._smdevice.volume_mute(mute)
+
+    def play_media(self, media_type, media_id, **kwargs):
+        """Play the specified media."""
+        pass
+
+    def select_source(self, source):
+        """Switch to the selected internet radio preset."""
+        self._smdevice.play_preset(self._sources_reverse[source])
+        self.schedule_update_ha_state()
+
+    def set_shuffle(self, shuffle):
+        """Enable/disable shuffle play."""
+        self._smdevice.set_shuffle(shuffle)
+
+    def set_volume_level(self, volume):
+        """Set the current volume level. This only works in pre-amp mode."""
+        volume = int(float(self._volume_max) * float(volume))
+        self._smdevice.set_volume(volume)
+
+    def turn_off(self):
+        """Switch the device OFF (default) or into IDLE mode."""
+        _LOGGER.debug("Powering off (%s)", self._power_off_cmd)
+        self._smdevice.power_off(power_state=self._power_off_cmd)
+        self._pwstate = STATE_OFF
+
+    def turn_on(self):
+        """Turn the device on if it's in IDLE (network standby) mode."""
+        if self._smdevice.get_power_state() == 'idle':
+            self._smdevice.power_on()
+            self._pwstate = STATE_ON
+
+    @staticmethod
+    def _to_seconds(timespec):
+        """Helper function to translate a h:mm:ss time spec into seconds."""
```

Comment:
Can't we use a date time helper instead of defining this method here? Seems like a common operation to do.

Reply:
Are you asking or suggesting? Is there already such a helper I could use?

Reply:
First try our dt util. See comment above.

Otherwise go for the datetime module in the standard library.
https://docs.python.org/3.6/library/datetime.html#strftime-strptime-behavior